main/src/ca/uwaterloo/flix/language/ast/WeededAst.scala

```diff
@@ -133,6 +133,8 @@ object WeededAst {
 
     case class ForEachYield(frags: List[ForFragment], exp: Expr, loc: SourceLocation) extends Expr
 
+    case class LetMatch(pat: Pattern, mod: Ast.Modifiers, tpe: Option[Type], exp1: Expr, exp2: Expr, loc: SourceLocation) extends Expr
```

Comment:
What kind of ordering are we using here?

Reply:
Can we try to keep everything sorted in the same way, following WeededAst? :)

Reply:
It already does, right?
I think Weeder was wrong before I began this (LetMatch comes after ForEachYield in Weeder and Ast, but Tuple is next in Ast whereas LetRecDef is next in Weeder)

Reply:
Anyway, we can change this after :)